src/GitVersion.Core.Tests/IntegrationTests/HotfixBranchScenarios.cs

```diff
@@ -167,7 +167,7 @@ public void FeatureOnHotfixFeatureBranchDeleted()
         fixture.Checkout(hotfix451);
         fixture.MergeNoFF(featureBranch); // commit 2
         fixture.Repository.Branches.Remove(featureBranch);
-        fixture.AssertFullSemver("4.5.1-beta.2", configuration);
+        fixture.AssertFullSemver("4.5.1-beta.3", configuration);
```

Comment:
Why did this go from `4.5.1-beta.2` to `4.5.1-beta.3`?

Reply:
Hmm very good question. I think it has something to do with the logic in GitVersionVariables.cs which you know will be changed in PR #2347. I have checked the source of the resulting base version and it looks okay for me:

![image](https://user-images.githubusercontent.com/56404113/229288034-0eee794a-2091-4ae1-b4f4-345f90f61879.png)

![image](https://user-images.githubusercontent.com/56404113/229287921-dee9a020-a7ed-403e-af86-3bada6caa9fb.png)

![image](https://user-images.githubusercontent.com/56404113/229287992-45b3dcbe-967a-4d95-91fd-ac321c2d9a3e.png)

Reply:
But the `.3` in `4.5.1-beta.3` indicates that there's 3 commits since the version source, but it is only 2 indicated by the previous `.2` as in `4.5.1-beta.2`. So this feels like a bug to me? Why does it count to 3 and not 2?

Reply:
If you take a look to the following figure:

![image](https://user-images.githubusercontent.com/56404113/229288034-0eee794a-2091-4ae1-b4f4-345f90f61879.png)

then you see that GitVersion uses the highest version (4.5.1). If there are ambiguous results (like in this case) the oldest commit as a source will be taken. Here it is the commit with the identifier bf7a8f7f. Thus if you have this in mind `4.5.1-beta.3` is totally correct. See yor comment here https://github.com/GitTools/GitVersion/issues/2394#issuecomment-1062170487